src/widget.cpp

```diff
@@ -1974,6 +1974,66 @@ int Scrollbar::GetScrolledRowFromWidget(int clickpos, const Window * const w, in
 	return (pos >= this->GetCount()) ? INT_MAX : pos;
 }
 
+/**
+ * Update the given list position as if it were on this scroll bar when the given key_code was pressed.
```

Comment:
This took me several tries to understand but I can't think of a better way to word it either.